src/NuGet.Core/NuGet.Commands/ContentFiles/ContentFileUtils.cs

```diff
@@ -0,0 +1,286 @@
+
+                    // Check if the nuspec group include/exclude patterns apply to the file
+                    var virtualDirectory = new VirtualFileProvider(new List<string>() { relativePath });
+                    var globbingDirectory = new FileProviderGlobbingDirectory(
+                        virtualDirectory,
+                        fileInfo: rootDirectory,
+                        parent: null);
+
+                    // Currently Matcher only returns the file name not the full path, each file must be
+                    // check individually.
+                    var matchResults = matcher.Execute(globbingDirectory);
+
+                    if (matchResults.Files.Any())
+                    {
+                        entryMappings[file].Add(filesEntry);
+                    }
+                }
+            }
+
+            // Create lock file entries for each item in the contentFiles folder
+            foreach (var file in entryMappings.Keys)
+            {
+                // defaults
+                var action = PackagingConstants.ContentFilesDefaultBuildAction;
+                var copyToOutput = false;
+                var flatten = false;
+
+                // apply each entry
+                // entries may not have all the attributes, if a value is null 
+                // ignore it and continue using the previous value.
+                foreach (var filesEntry in entryMappings[file])
+                {
+                    if (!string.IsNullOrEmpty(filesEntry.BuildAction))
+                    {
+                        action = filesEntry.BuildAction;
+                    }
+
+                    if (filesEntry.CopyToOutput.HasValue)
+                    {
+                        copyToOutput = filesEntry.CopyToOutput.Value;
+                    }
+
+                    if (filesEntry.Flatten.HasValue)
+                    {
+                        flatten = filesEntry.Flatten.Value;
+                    }
+                }
+
+                // Add attributes to the lock file item
+                var lockFileItem = new LockFileItem(file);
+
+                // Do not write out properties for _._
+                if (!file.EndsWith("/_._", StringComparison.Ordinal))
+                {
+                    // normalize and validate the build action name
+                    // the action will be null if the string was unrecognized
+                    var normalizedAction = GetBuildActionOrNull(action);
+
+                    if (normalizedAction == null)
+                    {
+                        // Throw an error containing the package identity, invalid action, and file where it occurred.
+                        var message = Strings.FormatError_UnknownBuildAction(nuspec.GetIdentity(), action, file);
+                        throw new PackagingException(message);
+                    }
+
+                    lockFileItem.Properties.Add("buildAction", normalizedAction);
```

Comment:
I don't think there would be a benefit, these are only used once and the names are specific to the lock file.